File: Lecture09/AsyncTask/app/src/main/java/com/codingblocks/asynctask/MainActivity.java
package com.codingblocks.asynctask;

import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = "ASYNC";

    Button button, button2;
    TextView tvResult;
    EditText etWaitTime;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button = (Button) findViewById(R.id.button);
        button2 = (Button) findViewById(R.id.button2);
        tvResult = (TextView) findViewById(R.id.tvResult);
        etWaitTime = (EditText) findViewById(R.id.etWaitTime);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // wait 10 sec
                // tvResult.setText("DONE");
                NSecWaitTask myTask = new NSecWaitTask(tvResult);
                myTask.execute(Integer.valueOf(etWaitTime.getText().toString()));

            }
        });
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                tvResult.setText("OTHER STUFF");

            }
        });

    }

    class NSecWaitTask extends AsyncTask<Integer, Float, String> {

        private TextView tvToChange;

        public NSecWaitTask(TextView tvToChange) {
            this.tvToChange = tvToChange;
        }

        @Override
        protected void onPreExecute() {
            tvToChange.setText("READY");
            super.onPreExecute();
        }

        @Override
        protected String doInBackground(Integer... params) {

            for (int i = 0; i < (params[0]) * 2; i++) {
                long startTime = System.currentTimeMillis();

                while(System.currentTimeMillis() < startTime + 500);

                publishProgress (((float) i) / 2);
            }


            return "BOOM";
        }

        @Override
        protected void onProgressUpdate(Float... values) {

            tvToChange.setText(String.valueOf(values[0]));

            super.onProgressUpdate(values);
        }

        @Override
        protected void onPostExecute(String s) {

            tvToChange.setText(s);
            super.onPostExecute(s);
        }
    }


}

File: Lecture09/AsyncTask/app/src/main/java/com/codingblocks/asynctask/MainActivity.kt
package com.codingblocks.asynctask

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        var myTv : TextView = findViewById(R.id.tvResult) as TextView;


        button.setOnClickListener {
            // wait 10 sec
            // tvResult.setText("DONE");
            val myTask = NSecWaitTask(tvResult)

            myTask.execute(Integer.valueOf(etWaitTime.text.toString()))
        }
        button2.setOnClickListener {
            tvResult.text = "SOME TEXT";

            var myStr: String = tvResult.text as String;

//            myTv!!.setText("SOMETHING");

        }


    }

    companion object {
        val TAG = "ASYNC"
    }


}

